pkg/reconciler/internal/hook/hook.go

```diff
@@ -43,6 +43,8 @@ func NewDependentResourceWatcher(c controller.Controller, rm meta.RESTMapper, ca
 	return &dependentResourceWatcher{
 		controller: c,
 		restMapper: rm,
+		cache:      cache,
+		scheme:     *scheme,
```

Comment:
If possible you should hold on to the pointer, not copy by value here with a dereference